### ISSUE TITLE
uloz.to download has failed

### ISSUE THREAD

**Issue by ririko5834** [pyload/pyload#4180]
### Description

I tried to download some file from uloz.to but once I add it to queue it fails.

### Debug log

```

```

```
[2022-07-30 12:38:58]  INFO                pyload  *** Welcome to pyLoad 0.5.0 ***
[2022-07-30 12:39:00]  DEBUG               pyload  Indexing plugins...
[2022-07-30 12:39:01]  INFO                pyload  ADDON ExternalScripts: Activated scripts in folder `download_finished`: extractmp3.sh
[2022-07-30 12:39:01]  INFO                pyload  Activated plugins: ClickNLoad, ExternalScripts, TORRENT, UnSkipOnFail, UserAgentSwitcher
[2022-07-30 12:39:01]  INFO                pyload  Deactivate plugins: AndroidPhoneNotify, AntiCaptcha, AntiStandby, AntiVirus, AppriseNotify, BypassCaptcha, Captcha9Kw, Checksum, CloudFlareDdos, DeathByCaptcha, DeleteFinished, DiscordNotifier, DownloadScheduler, ExpertDecoders, ExtractArchive, HotFolder, IRC, ImageTyperz, JustPremium, LinkFilter, LogMarker, MergeFiles, MultiHome, PushBullet, PushOver, RestartFailed, SkipRev, TransmissionRPC, UpdateManager, WindowsPhoneNotify, XFileSharing, XMPP
[2022-07-30 12:39:02]  DEBUG               pyload  Starting core...
[2022-07-30 12:39:02]  DEBUG               pyload  Debug level: TRACE
[2022-07-30 12:39:02]  DEBUG               pyload  Setup language...
[2022-07-30 12:39:02]  DEBUG               pyload  Setup permissions...
[2022-07-30 12:39:02]  INFO                pyload  User directory: /home/pi/.pyload
[2022-07-30 12:39:02]  INFO                pyload  Cache directory: /tmp/pyLoad
[2022-07-30 12:39:02]  INFO                pyload  Storage directory: /home/pi/Downloads/pyLoad
[2022-07-30 12:39:02]  INFO                pyload  Storage free space: 6.03 GiB
[2022-07-30 12:39:02]  DEBUG               pyload  Setup network...
[2022-07-30 12:39:02]  INFO                pyload  Activating accounts...
[2022-07-30 12:39:02]  INFO                pyload  Activating plugins...
[2022-07-30 12:39:02]  DEBUG               pyload  ADDON ExternalScripts: No script found under folder `pyload_start`
[2022-07-30 12:39:02]  WARNING             pyload  ADDON TORRENT: torrents / magnets are not associated with any plugin
[2022-07-30 12:39:02]  INFO          pyload.webui  Starting webserver: http://0.0.0.0:8000
[2022-07-30 12:39:02]  DEBUG               pyload  *** pyLoad is up and running ***
[2022-07-30 12:39:06]  DEBUG               pyload  ADDON ClickNLoad: Backend found on http://127.0.0.1:8000
[2022-07-30 12:39:06]  INFO                pyload  ADDON ClickNLoad: Proxy listening on 0.0.0.0:9666
[2022-07-30 12:39:19]  INFO                pyload  Added package ulozto movie1 containing 1 links
[2022-07-30 12:39:19]  INFO                pyload  Download starts: https://uloz.to/file/NQvgyrl4WATz/morska-prisera-novinka-2022-cz-dabing-mkv#!ZJWxAQR2Zwp2BTWxBTWxMwDkZmV3LKcHDIqXFyp5GzkgAQL4AN==
[2022-07-30 12:39:19]  DEBUG               pyload  ADDON ExternalScripts: No script found under folder `download_preparing`
[2022-07-30 12:39:19]  DEBUG               pyload  ADDON UserAgentSwitcher: Setting connection timeout to 60 seconds
[2022-07-30 12:39:19]  DEBUG               pyload  ADDON UserAgentSwitcher: Setting maximum redirections to 10
[2022-07-30 12:39:19]  DEBUG               pyload  ADDON UserAgentSwitcher: Use custom user-agent string `Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:87.0) Gecko/20100101 Firefox/87.0`
[2022-07-30 12:39:19]  DEBUG               pyload  DOWNLOADER UlozTo[583]: Plugin version: 1.50
[2022-07-30 12:39:19]  DEBUG               pyload  DOWNLOADER UlozTo[583]: Plugin status: testing
[2022-07-30 12:39:19]  WARNING             pyload  DOWNLOADER UlozTo[583]: Plugin may be unstable
[2022-07-30 12:39:19]  INFO                pyload  DOWNLOADER UlozTo[583]: Processing url: https://uloz.to/file/NQvgyrl4WATz/morska-prisera-novinka-2022-cz-dabing-mkv#!ZJWxAQR2Zwp2BTWxBTWxMwDkZmV3LKcHDIqXFyp5GzkgAQL4AN==
[2022-07-30 12:39:19]  INFO                pyload  DOWNLOADER UlozTo[583]: Looking for direct download link...
[2022-07-30 12:39:19]  DEBUG               pyload  DOWNLOADER UlozTo[583]: LOAD URL https://ulozto.net/file/NQvgyrl4WATz/morska-prisera-novinka-2022-cz-dabing-mkv#!ZJWxAQR2Zwp2BTWxBTWxMwDkZmV3LKcHDIqXFyp5GzkgAQL4AN== | get={} | post={} | ref=True | cookies=True | just_header=True | decode=True | multipart=False | redirect=True | req=None
[2022-07-30 12:39:19]  INFO                pyload  DOWNLOADER UlozTo[583]: Direct download link not found
[2022-07-30 12:39:19]  DEBUG               pyload  DOWNLOADER UlozTo[583]: LOAD URL https://ulozto.net/file/NQvgyrl4WATz/morska-prisera-novinka-2022-cz-dabing-mkv#!ZJWxAQR2Zwp2BTWxBTWxMwDkZmV3LKcHDIqXFyp5GzkgAQL4AN== | get={} | post={} | ref=False | cookies=True | just_header=False | decode=True | multipart=False | redirect=True | req=None
[2022-07-30 12:39:20]  INFO                pyload  DOWNLOADER UlozTo[583]: Checking for link errors...
[2022-07-30 12:39:20]  INFO                pyload  DOWNLOADER UlozTo[583]: No errors found
[2022-07-30 12:39:20]  INFO                pyload  DOWNLOADER UlozTo[583]: Grabbing link info...
[2022-07-30 12:39:20]  DEBUG               pyload  DOWNLOADER UlozTo[583]: Link info: {'name': 'Morska prisera NOVINKA 2022 CZ dabing.mkv', 'hash': {}, 'pattern': {'ID': 'file/NQvgyrl4WATz', 'N': 'Morska prisera NOVINKA 2022 CZ dabing.mkv'}, 'size': 0, 'status': 2, 'url': 'https://ulozto.net/file/NQvgyrl4WATz/morska-prisera-novinka-2022-cz-dabing-mkv#!ZJWxAQR2Zwp2BTWxBTWxMwDkZmV3LKcHDIqXFyp5GzkgAQL4AN=='}
[2022-07-30 12:39:20]  DEBUG               pyload  DOWNLOADER UlozTo[583]: Previous link info: {}
[2022-07-30 12:39:20]  INFO                pyload  DOWNLOADER UlozTo[583]: Link name: Morska prisera NOVINKA 2022 CZ dabing.mkv
[2022-07-30 12:39:20]  INFO                pyload  DOWNLOADER UlozTo[583]: Link size: N/D
[2022-07-30 12:39:20]  INFO                pyload  DOWNLOADER UlozTo[583]: Link status: online
[2022-07-30 12:39:20]  INFO                pyload  DOWNLOADER UlozTo[583]: Processing as free download...
[2022-07-30 12:39:20]  DEBUG               pyload  DOWNLOADER UlozTo[583]: LOAD URL https://ulozto.net/download-dialog/free/download?fileSlug=NQvgyrl4WATz | get={} | post={} | ref=True | cookies=True | just_header=False | decode=True | multipart=False | redirect=True | req=None
[2022-07-30 12:39:20]  DEBUG               pyload  DOWNLOADER UlozTo[583]: inputs.keys = dict_keys(['timestamp', 'salt', 'hash', 'captcha_type', 'captcha_value', '_do'])
[2022-07-30 12:39:20]  DEBUG               pyload  DOWNLOADER UlozTo[583]: Using "new" version
[2022-07-30 12:39:20]  DEBUG               pyload  DOWNLOADER UlozTo[583]: LOAD URL https://ulozto.net/reloadXapca.php | get={'rnd': 1659177560589} | post={} | ref=True | cookies=True | just_header=False | decode=True | multipart=False | redirect=True | req=None
[2022-07-30 12:39:20]  DEBUG               pyload  DOWNLOADER UlozTo[583]: xapca: {"image":"https:https:\/\/xapca.uloz.to\/f281b7b70f02321b31c940e068dcc341c1755eee\/image.jpg","sound":"https:https:\/\/xapca.uloz.to\/f281b7b70f02321b31c940e068dcc341c1755eee\/sound.mp3","hash":"f281b7b70f02321b31c940e068dcc341c1755eee","timestamp":1659177560,"salt":"1540"}
[2022-07-30 12:39:20]  DEBUG               pyload  ANTICAPTCHA UlozTo[583]: BaseCaptcha | LOAD URL https:https://xapca.uloz.to/f281b7b70f02321b31c940e068dcc341c1755eee/image.jpg | get={} | post={} | ref=False | cookies=True | just_header=False | decode=False | multipart=False | redirect=True | req=<pyload.core.network.browser.Browser object at 0xa9c2786c>
[2022-07-30 12:39:20]  DEBUG               pyload  ADDON ExternalScripts: No script found under folder `download_processed`
[2022-07-30 12:39:20]  DEBUG               pyload  ADDON ExternalScripts: No script found under folder `package_processed`
[2022-07-30 12:39:20]  WARNING             pyload  Download failed: Morska prisera NOVINKA 2022 CZ dabing.mkv | (3, '')
Traceback (most recent call last):
  File "/home/pi/.local/lib/python3.7/site-packages/pyload/core/threads/download_thread.py", line 56, in run
    pyfile.plugin.preprocessing(self)
  File "/home/pi/.local/lib/python3.7/site-packages/pyload/plugins/base/hoster.py", line 291, in preprocessing
    return self._process(*args, **kwargs)
  File "/home/pi/.local/lib/python3.7/site-packages/pyload/plugins/base/downloader.py", line 103, in _process
    self.process(self.pyfile)
  File "/home/pi/.local/lib/python3.7/site-packages/pyload/plugins/base/simple_downloader.py", line 282, in process
    self.handle_free(pyfile)
  File "/home/pi/.local/lib/python3.7/site-packages/pyload/plugins/downloaders/UlozTo.py", line 162, in handle_free
    captcha_value = self.captcha.decrypt(data['image'])
  File "/home/pi/.local/lib/python3.7/site-packages/pyload/plugins/base/captcha.py", line 61, in decrypt
    req=req or self.pyfile.plugin.req,
  File "/home/pi/.local/lib/python3.7/site-packages/pyload/plugins/base/plugin.py", line 226, in load
    decode is True,
  File "/home/pi/.local/lib/python3.7/site-packages/pyload/core/network/browser.py", line 127, in load
    return self.http.load(*args, **kwargs)
  File "/home/pi/.local/lib/python3.7/site-packages/pyload/core/network/http/http_request.py", line 293, in load
    self.c.perform()
pycurl.error: (3, '')
[2022-07-30 12:39:20]  INFO                pyload  Debug Report written to /tmp/pyLoad/debug_UlozTo_2022-07-30_12-39-20.zip
[2022-07-30 12:39:20]  DEBUG               pyload  ADDON ExternalScripts: No script found under folder `download_failed`
[2022-07-30 12:39:20]  INFO                pyload  ADDON UnSkipOnFail: Looking for skipped duplicates of: Morska prisera NOVINKA 2022 CZ dabing.mkv (pid:11)
[2022-07-30 12:39:20]  INFO                pyload  ADDON UnSkipOnFail: No duplicates found
[2022-07-30 12:39:20]  DEBUG               pyload  ADDON ExternalScripts: No script found under folder `all_downloads_processed`
[2022-07-30 12:39:20]  DEBUG               pyload  All downloads processed
```

**Comment by ririko5834** [comment]
it even didnt start downloading 
![image](https://user-images.githubusercontent.com/64729293/181906744-f981a237-8ae4-481e-966f-7547bae711ec.png)

**Comment by ririko5834** [comment]
@vuolter @GammaC0de You commited that that download plugin, can you help me?

**Comment by danoslas** [comment]
@ririko5834  Hi, I've already fixed this issue. My changes were merged in pull request #4176.

**Comment by GammaC0de** [comment]
@ririko5834 
```
[2022-07-30 12:39:19]  DEBUG               pyload  DOWNLOADER UlozTo[583]: Plugin version: 1.50
```
As @ririko5834 already said, your `UlozTo` plugin is not up-to-date, please update your installation:


```
pip install --upgrade https://github.com/pyload/pyload/archive/refs/heads/develop.zip
```
